lib/SILOptimizer/Utils/InstructionDeleter.cpp

```diff
@@ -63,6 +63,11 @@ static bool isScopeAffectingInstructionDead(SILInstruction *inst,
     return false;
   }
 
+  // Don't delete dead drop_deinit instruction.
```

Comment:
This comment just states the obvious. Can you add a comment explaining _why_ we don't delete `drop_deinit` instructions?